day04/spec/unit/main_spec.rb

```diff
@@ -0,0 +1,110 @@
+require_relative '../../lib/main.rb'
+
+require 'pry'
+RSpec.describe Main do
+
+
+  let(:main) {
+    Main.new
+  }
+
+  let(:product1){
+    Product.new("Pen", 1.2, 1)
+  }
+
+  let(:product2){
+    Product.new("Notebook", 2, 1)
+  }
+
+  let(:product3){
+    Product.new("Pencil", 1.5, 1)
+  }
+
+  let(:basket) do [
+    product1,
+    Product.new("Notebook", 2, 1)
+  ] end
+
+
+  context ".choose_product" do
+
```

Comment:
When you describe method use `.` to describe class method and `#` to describe instance method. In your case because in main.rb you have instance methods, you should change `.` -> `#`.

http://betterspecs.org/#describe